services/gemma-local/model_manager.py

```
@@ -474,20 +555,98 @@
+            if mem_percent > 30:
+                logger.warning(f"High process memory usage: {mem_percent:.1f}%")
+
+        except Exception as e:
+            logger.debug(f"Failed to log memory usage: {e}")
+
+    def _check_memory_pressure(self) -> bool:
+        """Check if system is under memory pressure."""
+        try:
+            system_mem = psutil.virtual_memory()
+            return system_mem.percent > 80
+        except:
```

Ruff (0.13.1)

Do not use bare `except`

(E722)